sov-modules/sov-modules-api/src/lib.rs

```diff
@@ -34,22 +34,25 @@ impl AsRef<[u8]> for Address {
     }
 }
 
-impl<'a> TryFrom<&'a [u8]> for Address {
-    type Error = anyhow::Error;
-
-    fn try_from(addr: &'a [u8]) -> Result<Self, Self::Error> {
-        Ok(Self {
-            addr: addr.to_vec(),
-        })
-    }
-}
-
 impl AddressTrait for Address {}
 
 /// Default implementation of AddressTrait for the module system
 #[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq, Clone, Eq)]
 pub struct Address {
```

Comment:
Looks great. I think we can remove the custom address from the demo-rollup once this lands too